Assets/SEE/UI/Window/ConsoleWindow/ConsoleWindow.cs

```diff
@@ -0,0 +1,691 @@
+            }
+        }
+
+        /// <summary>
+        /// Clears all messages.
+        /// </summary>
+        public static void ClearMessages()
+        {
+            messages.Clear();
+            MessagesCleared?.Invoke();
+        }
+
+        /// <summary>
+        /// Adds a new channel.
+        /// </summary>
+        /// <param name="channel">The channel name.</param>
+        /// <param name="icon">The channel icon.</param>
+        public static void AddChannel(string channel, char icon)
+        {
+            channels[channel] = new Channel(channel, icon);
+            ChannelChanged?.Invoke();
+        }
+
+        /// <summary>
+        /// Adds a level to a channel.
+        /// </summary>
+        /// <param name="channel">The channel name.</param>
+        /// <param name="level">The level name.</param>
+        /// <param name="color">The level color.</param>
+        public static void AddChannelLevel(string channel, string level, Color color)
+        {
+            if (channels.TryGetValue(channel, out Channel c))
+            {
+                c.Levels[level] = new(level, color, true);
+                ChannelChanged?.Invoke();
+            }
+            else
+            {
+                Debug.LogWarning($"Channel {channel} doesn't exist");
+            }
+        }
+
+        /// <summary>
+        /// Sets whether 
+        /// </summary>
+        /// <param name="channel"></param>
+        /// <param name="level"></param>
+        /// <param name="enabled"></param>
+        public static void SetChannelLevelEnabled(string channel, string level, bool enabled)
+        {
+            if (channels.TryGetValue(channel, out Channel c)) {
+                if (c.Levels.TryGetValue(level, out ChannelLevel l))
+                {
+                    l.enabled = enabled;
+                    ChannelChanged?.Invoke();
+                } else
+                {
+                    Debug.LogWarning($"Level {channel} doesn't exist for channel {channel}.");
+                }
+            } else
+            {
+                Debug.LogWarning($"Channel {channel} doesn't exist.");
+            }
+        }
+
+
+        /// <summary>
+        /// Returns the message index to which a new message should be appended.
+        /// -1 if the a new message should be created.
+        /// </summary>
+        /// <param name="channel">The channel.</param>
+        /// <param name="level">The level.</param>
+        /// <returns></returns>
+        private static int AppendTo(string channel, string level)
+        {
+            for (int i=messages.Count-1; i>=0; i--)
+            {
+                Message message = messages[i];
+                if (message.Channel == channel && message.Level == level)
+                {
+                    if (!message.Text.EndsWith('\n'))
+                    {
+                        return i;
+                    }
+                }
+            }
+            return -1;
```

Comment:
Can be simplified:
```suggestion
            return messages.Reverse()
                           .FirstOrDefault(m => m.Channel == channel && m.Level == level && !message.Text.EndsWith('\n'), -1);
```

Reply:
Sorry, I was referring to [this `Reverse` method in `IEnumerable`](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.reverse?view=net-8.0), I didn't know `List` had an overload with a void return type. I think doing `messages.AsEnumerable().Reverse()` should fix it.

Reply:
There are further problems with using `FirstOrDefault`:
1. `FirstOrDefault` only takes 1 argument (your suggestions passes 2 parameters)
2. `FirstOrDefault` returns the found element (and not the index)